File: Python/demo_scAIG_V.py
import numpy as np
import time
from scAIG_C import scAIG_C
import matplotlib.pyplot as plt
import warnings
import random
from scAIG_V import scAIG_V
import warnings
from sklearn.metrics.cluster import adjusted_rand_score as ari
from sklearn import metrics
warnings.filterwarnings("ignore")
# seed = 42
# random.seed(seed)
Datasets = ['Yan']
n_datasets = len(Datasets)
Res_Visual_AIG = []
for id in range(1):
    data = Datasets[id]
    print(data)
    fea = np.load(data+'_cell.npy')
    gnd = np.load(data+'_label.npy')
    t1 = time.time()
    [Y,grp,idx_cell] = scAIG_V(fea)
    t2 = time.time()
    labels = gnd-np.min(gnd)+1
    labels = labels[idx_cell]
    labels = labels-1
    labels = labels.reshape(grp.shape)
    ARI = ari(labels,grp)
    for ii in range(np.max(labels)):
        plt.scatter(Y[labels==ii,0],Y[labels==ii,1])
    plt.show()
    si = metrics.silhouette_score(Y,labels)
    print('ARI=%2,4f.\n',ARI)
    print('Embedding CPU Time = %2,4f.\n',t2-t1)
    print('Silhouette Coefficient = %2,4f.\n',si)

File: Matlab/demo_scAIG_V.m
%% Load data.
load('Colors.mat');
seed = 1;
rng(seed);
Datasets = {...
    'yann';
    'Goolam';
    'Deng';
    'Darmanis';
    'Usoskin';
    'Xin';
    'Muraro';
    'Lake';
    };
n_datasets = length(Datasets); Res_Visual_AIG = cell(n_datasets,4);

for id = 1:8
    data = Datasets{id,1}; disp(data); eval(['load ' data]); options = [];
    [Y,grp,idx_cell] = scAIG_V(fea,options);
    labels = gnd - min(gnd) + 1;
    labels = labels(idx_cell);
    ARI = calARI(grp,labels);
    fprintf('ARI = %2.4f.\n',ARI);
    
    subplot(2,4,id);
    for ii = 1: length(unique(labels))
        hold on;
        scatter(Y(labels == ii,1), Y(labels == ii,2),10,'filled','MarkerFaceColor',colors(ii,:),'MarkerEdgeColor',colors(ii,:)); pause(0.1),
    end
    
    s = silhouette(Y,labels); measilhouette = mean(s);
    title([Datasets{id},', Silhouette = ', num2str(round(100*mean(s))/100)]);
    
    xlim([min(Y(:,1))-0.05,max(Y(:,1))+0.05]);
    ylim([min(Y(:,2))-0.05,max(Y(:,2))+0.05]);
    
    Res_Visual_AIG{id,1} = data;    
    Res_Visual_AIG{id,2} = Y;
    Res_Visual_AIG{id,3} = labels;
    Res_Visual_AIG{id,4} = measilhouette;
    pause(0.1),
end



